### PR TITLE
Scripts/Grizzlyhills: Spell: Infected Worgen Bite - Apply Spellscript and Transformation

### LINKED ISSUE #12165: [3.3.5a][Spell] Worgen's Call
rev c06dc7d // newest DB
spells:
Infected Worgen Bite : 53094
Worgen's Call : 53095

Situation:
You get three stacks of Infected Worgen Bite, but you don't get transformed into a worgen.

How it should be:
After 3 stacks, the aura  53094 should be removed and therefore you get the aura 53095.

proof:
By dreadblood (14,112 – 3·12·124) on 2009/08/25 (Patch 3.2.0)		
personally i like to get the 3 debuffs to become a worgen.. its fun :)
http://www.wowhead.com/spell=53095#comments (first)

I tried this, but it does not work.

```diff
class spell_worgens_call : public SpellScriptLoader
{
    public:
        spell_worgens_call() : SpellScriptLoader("spell_worgens_call") { }

        class spell_worgens_call_SpellScript : public SpellScript
        {
            public:
                PrepareSpellScript(spell_worgens_call_SpellScript)

            uint32 worgen;

            bool Load()
            {
                worgen = sSpellMgr->GetSpellIdForDifficulty(SPELL_WORGEN_BITE, GetCaster());
                if (!sSpellMgr->GetSpellInfo(worgen))
                    return false;

                return true;
            }

            void HandleScriptEffect(SpellEffIndex /*effIndex*/)
            {
                if (Unit* target = GetHitUnit())
                {
                    if (Aura* pAura = target->GetAura(worgen))
                    {
                        if (pAura->GetStackAmount() >= 3)
                        {
                            if (target->GetDummyAuraEffect(SPELLFAMILY_GENERIC, 2206, EFFECT_1))
                                target->CastSpell(target, SPELL_WORGEN_CALL, true);

                            target->RemoveAurasDueToSpell(worgen);
                        }
                    }
                }
            }

            void Register()
            {
                OnEffectHitTarget += SpellEffectFn(spell_worgens_call_SpellScript::HandleScriptEffect, EFFECT_1, SPELL_EFFECT_SCRIPT_EFFECT);
            }
        };

        SpellScript* GetSpellScript() const
        {
            return new spell_worgens_call_SpellScript();
        }
};
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/2399499-3-3-5a-spell-worgen-s-call?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1310&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Lumber-Jack-2**
BY untaught

closes #12165 

before: http://pokit.org/get/?ac650613f71168b8de205855c55a7232.jpg
after: http://pokit.org/get/?d7b60982d8efc03ee52567b33a7f9ab6.jpg

### CODE REVIEW

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 1]
```diff
@@ -816,6 +816,44 @@ class spell_shredder_delivery : public SpellScriptLoader
...
+                    if (target->GetAura(SPELL_INFECTED_WORGEN_BITE)->GetStackAmount() == 3)
+                    {
+                        target->RemoveAurasDueToSpell(SPELL_INFECTED_WORGEN_BITE);
+                        target->CastSpell(target, SPELL_WORGENS_CALL, true);
+                    }
+            }
+            
+            void Register() override
+            {
+                AfterEffectApply += AuraEffectApplyFn(spell_infected_worgen_bite_AuraScript::HandleAfterEffectApply, EFFECT_ALL, SPELL_AURA_PERIODIC_DAMAGE, AURA_EFFECT_HANDLE_CHANGE_AMOUNT_MASK);
```
Comment (joschiwald) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: AfterEffectApply += AuraEffectApplyFn(spell_infected_worgen_bite_AuraScript::HandleAfterEffectApply, EFFECT_1, SPELL_AURA_PERIODIC_DAMAGE, AURA_EFFECT_HANDLE_REAPPLY);

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 2]
```diff
@@ -816,6 +816,44 @@ class spell_shredder_delivery : public SpellScriptLoader
...
+            PrepareAuraScript(spell_infected_worgen_bite_AuraScript);
+            
+            void HandleAfterEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                Unit* target = GetTarget();
+                if (target->GetTypeId() == TYPEID_PLAYER)
+                    if (target->GetAura(SPELL_INFECTED_WORGEN_BITE)->GetStackAmount() == 3)
+                    {
+                        target->RemoveAurasDueToSpell(SPELL_INFECTED_WORGEN_BITE);
+                        target->CastSpell(target, SPELL_WORGENS_CALL, true);
```
Comment (joschiwald) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]:
```c++
if (GetStackAmount() == GetSpellInfo()->StackAmount)
{
    Remove();
    target->CastSpell(target, SPELL_WORGENS_CALL, true);
}
```

File: sql/updates/world/2015_02_06_14_world.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+--  assign spellscript for Spell: 'Infected Worgen Bite'
+DELETE FROM `spell_linked_spell` WHERE `spell_trigger`=53095;
+INSERT INTO `spell_linked_spell` (`spell_trigger`, `spell_effect`, `type`, `comment`) VALUES
+(53095,-53094,2,'');
```
Comment (joschiwald) [inline sql/updates/world/2015_02_06_14_world.sql]: immunity needed here? and a comment will be great

**Comment by Rushor** [comment]
thx, with these changes it's still working: http://pokit.org/get/?b316c0c96bb8ff7e99d8448798bbe2aa.jpg

File: src/server/scripts/Northrend/zone_grizzly_hills.cpp [hunk 4]
```diff
@@ -816,6 +816,44 @@ class spell_shredder_delivery : public SpellScriptLoader
...
+        {
+            PrepareAuraScript(spell_infected_worgen_bite_AuraScript);
+            
+            void HandleAfterEffectApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+            {
+                Unit* target = GetTarget();
+                if (target->GetTypeId() == TYPEID_PLAYER)
+                    if (GetStackAmount() == GetSpellInfo()->StackAmount)
+                    {
+                        Remove();
```
Comment (Gooyeth) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: The aura is removed in spell_linked_spell.

Comment (Rushor) [inline src/server/scripts/Northrend/zone_grizzly_hills.cpp]: nope that's only an immunity

**Comment by Nyeriah** [comment]
Anything left about this one or is it safe to be merged?

**Comment by Rushor** [comment]
well someone must confirm, if the player should be immune to the debuff
(this part: (53095,-53094,2,'Infected Worgen Bite - Worgen\'s Call Immunity');)

i tried to test this on retail, but these creatures do not hit level 100 player, i cannot test it.

if someone wants to test it with a level 70-80 char: http://www.wowhead.com/spell=53094/infected-worgen-bite#used-by-npc

**Pull request merged**